governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/utils.ts

```diff
@@ -35,66 +35,89 @@ export const getProposalStatus = (
  */
-export const getInstructionsSummary = (options: {
+export const getInstructionsSummary = ({
+  instructions,
+  cluster,
+}: {
   instructions: MultisigInstruction[]
   cluster: PythCluster
-}) => {
-  const { instructions, cluster } = options
+}) =>
+  Object.entries(
+    getInstructionSummariesByName(
+      MultisigParser.fromCluster(cluster),
+      instructions
+    )
+  )
+    .map(([name, summaries = []]) => ({
+      name,
+      count: summaries.length ?? 0,
+      summaries,
+    }))
+    .toSorted(({ count }) => count)
 
-  return sortObjectByValues(
-    instructions.reduce((acc, instruction) => {
-      if (instruction instanceof WormholeMultisigInstruction) {
-        const governanceAction = instruction.governanceAction
-        if (governanceAction instanceof ExecutePostedVaa) {
-          const innerInstructions = governanceAction.instructions
-          innerInstructions.forEach((innerInstruction) => {
-            const multisigParser = MultisigParser.fromCluster(cluster)
-            const parsedInstruction = multisigParser.parseInstruction({
-              programId: innerInstruction.programId,
-              data: innerInstruction.data as Buffer,
-              keys: innerInstruction.keys as AccountMeta[],
-            })
-            acc[parsedInstruction.name] = (acc[parsedInstruction.name] ?? 0) + 1
-          })
-        } else if (governanceAction instanceof PythGovernanceActionImpl) {
-          acc[governanceAction.action] = (acc[governanceAction.action] ?? 0) + 1
-        } else if (governanceAction instanceof SetDataSources) {
-          acc[governanceAction.actionName] =
-            (acc[governanceAction.actionName] ?? 0) + 1
-        } else {
-          acc['unknown'] = (acc['unknown'] ?? 0) + 1
-        }
-      } else {
-        acc[instruction.name] = (acc[instruction.name] ?? 0) + 1
-      }
-      return acc
-    }, {} as Record<string, number>)
+const getInstructionSummariesByName = (
+  parser: MultisigParser,
+  instructions: MultisigInstruction[]
+) =>
+  Object.groupBy(
+    instructions.flatMap((instruction) =>
+      getInstructionSummary(parser, instruction)
+    ),
+    ({ name }) => name
   )
+
+const getInstructionSummary = (
+  parser: MultisigParser,
+  instruction: MultisigInstruction
+) => {
+  if (instruction instanceof WormholeMultisigInstruction) {
+    const { governanceAction } = instruction
+    if (governanceAction instanceof ExecutePostedVaa) {
+      return governanceAction.instructions.map((innerInstruction) =>
+        getTransactionSummary(parser, innerInstruction)
+      )
+    } else if (governanceAction instanceof PythGovernanceActionImpl) {
+      return [{ name: governanceAction.action } as const]
+    } else if (governanceAction instanceof SetDataSources) {
+      return [{ name: governanceAction.actionName } as const]
+    } else {
+      return [{ name: 'unknown' } as const]
+    }
+  } else {
+    return [{ name: instruction.name } as const]
```

Comment:
When instruction is not a `WormholeMultisigInstruction`, you should also generate a summary if it's `addPublisher` or `delPublisher`

Reply:
Oh interesting, I derived this logic by looking at when we display more details for `addPublisher` and `delPublisher` -- in particular I looked at where leverage the mappings from keys to symbol & publisher name -- and I noticed this was only happening [in the component to render `WormholeMultisigInstruction` types](https://github.com/pyth-network/pyth-crosschain/blob/c0a4ab04ddc0c590f0dd5f699d3b88512ac0e88f/governance/xc_admin/packages/xc_admin_frontend/components/InstructionViews/WormholeInstructionView.tsx#L225-L236).  Does this mean we should update the details pages to expand that info for other instruction types too?

Reply:
Here's an example of a proposal where this comes into play: https://proposals.pyth.network/?tab=proposals&proposal=8zU6pr28Ta5zLwn5zPH2dWjHTAVGhg4VnjwpHeGALmFg

Reply:
ah yes, I missed it [here](https://github.com/pyth-network/pyth-crosschain/blob/c0a4ab04ddc0c590f0dd5f699d3b88512ac0e88f/governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals/Proposal.tsx#L633-L642)!  Apologies on the oversight.  I will update this to account for non-wormhole instructions.

Reply:
Updated!